pennylane/workflow/qnode.py

```diff
@@ -932,8 +933,12 @@ def construct(self, args, kwargs):  # pylint: disable=too-many-branches
         if old_interface == "auto":
             self.interface = qml.math.get_interface(*args, *list(kwargs.values()))
 
-        with qml.queuing.AnnotatedQueue() as q:
-            self._qfunc_output = self.func(*args, **kwargs)
+        # Before constructing the tape, we pass the device to the
+        # debugger to ensure they are compatible if there are any
+        # breakpoints in the circuit

```

Comment:
So is this just for checking compatibility? If so, do we need to use a context with an enter-exit just for checking validity?

Reply:
I think that's the key, we don't need to do validation checks if there are no breakpoints in the circuit. Furthermore, we use the device to execute measurements (see the following [PR](https://github.com/PennyLaneAI/pennylane/pull/5749)), so we do need the context.